### PR TITLE
feat(controller): add domain names for object storage

### PR DESCRIPTION
**Pull request by anda-ren**
## Description

recognize regex pattern for the object store domain in the header

client part:
https://github.com/star-whale/starwhale/pull/3048

## Test cases
image

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/configuration/security/ObjectStoreDomainDetectionFilter.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        Map<String, String> endpointEquivalentsMap = s3Config.getEndpointEquivalentsMap();
+        domainAliasMap = endpointEquivalentsMap.entrySet().stream().collect(Collectors.toMap(Entry::getKey, (entry) -> {
+            URI uri = null;
+            try {
+                uri = new URI(entry.getValue());
+            } catch (URISyntaxException e) {
+                throw new RuntimeException(e);
+            }
+            return Pattern.compile(uri.getHost().replace(".", "\\."));
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/configuration/security/ObjectStoreDomainDetectionFilter.java:59]: What if the host configured by the user is the same, but the sub-path is different? Can we directly let the OSS impl class support sign by name?

Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/configuration/security/ObjectStoreDomainDetectionFilter.java:59]: > What if the host configured by the user is the same, but the sub-path is different? Can we directly let the OSS impl class support sign by name?

If the host is the same while the sub-path is different, they are not equivalent

Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/configuration/security/ObjectStoreDomainDetectionFilter.java:59]: > What if the host configured by the user is the same, but the sub-path is different? Can we directly let the OSS impl class support sign by name?

If the OSS impl class support sign by name , every impl class has to do similar implementation which is code duplication indeed. And that breaks the original interface which will cause clients' code change

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**